lib/kino/livebook_output.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Kino.LivebookOutput do
+  @moduledoc """
+  A number of output formats supported by Livebook.
+  """
+
+  @typedoc """
+  Livebook cell output may be one of these values
+  and is rendered accordingly.
+  """
+  @type t ::
+          :ignored
+          # Regular text, adjacent such outputs can be treated as a whole
+          | binary()
+          # Standalone text block
+          | {:text, binary()}
+          # Vega-Lite graphic
+          | {:vega_lite_static, spec :: map()}
+          # Vega-Lite graphic with dynamic data
+          | {:vega_lite_dynamic, widget_process :: pid()}
```

Comment:
@josevalim One thing I'm not quite certain about is where to document the communication protocol. This output type is theoretically Kino-agnostic, because anyone could have a process that exposes Vega-Lite stuff, but currently the protocol is described [in kino](https://github.com/elixir-nx/kino/blob/22411b82e2d5a3d12d7c5738c86f01e958fc0257/lib/kino/vega_lite.ex#L25-L37).

Reply:
At least from documentation perspective, we could describe the protocol under a separate type like this:

```elixir
  @typedoc """
  Livebook cell output may be one of these values
  and is rendered accordingly.
  """
  @type t ::
          :ignored
          # Regular text, adjacent such outputs can be treated as a whole
          | binary()
          # Standalone text block
          | {:text, binary()}
          # Vega-Lite graphic
          | {:vega_lite_static, spec :: map()}
          # Vega-Lite graphic with dynamic data
          | {:vega_lite_dynamic, vega_lite_pid()}

  @typedoc """
  A process wrapping VegaLite graphic and allowing to
  dynamically send data to the clients.

  ## Communication protocol

  A client process should connect to the VegaLite process by sending:

      {:connect, pid}

  And expect the following reply:

      {:connect_reply, %{spec: map()}}

  The VegaLite process may then keep sending one of the following events:

      {:push, %{data: list(), dataset: binary(), window: non_neg_integer()}}
  """
  @type vega_lite_pid :: pid()
```

I'm not sure if this should live here on in `Livebook.Notebook.Cell.output` though. Thoughts?

Reply:
It is all "private" for now, so I think it is fine to document it here. You can also do this:

```elixir
| vega_lite_static()
```

And then:

```elixir
@doc ...
@type vega_lite_static() :: {:vega_lite_static, spec :: map()}
```

Reply:
I documented all types in this manner, hopefully not overengineered :see_no_evil: